opamp/observiq/observiq_client.go

```diff
@@ -193,16 +189,45 @@ func (c *Client) onErrorHandler(errResp *protobufs.ServerErrorResponse) {
 	c.logger.Error("Server returned an error response", zap.String("Error", errResp.GetErrorMessage()))
 }
 
-func (c *Client) onRemoteConfigHandler(_ context.Context, remoteConfig *protobufs.AgentRemoteConfig) (*protobufs.EffectiveConfig, bool, error) {
+func (c *Client) onMessageFuncHandler(ctx context.Context, msg *types.MessageData) {
+	switch {
+	// Fill in other message types as needed
+	case msg.RemoteConfig != nil:
+		if err := c.onRemoteConfigHandler(ctx, msg.RemoteConfig); err != nil {
+			c.logger.Error("Error while processing Remote Config Change", zap.Error(err))
+		}
+	}
```

Comment:
This doesn't really matter for this PR since we only have one message type we respond to, but should this be an if statement instead? 

Not sure if each message can only provide one "sub-message", or if it can have several at once, if that makes sense.

Reply:
That is a good question...I think we likely need to ask the BP team. If it's only one at a time then the switch makes sense if it's multiple then we need if statements. The BP team is still working on implementing this so I'll ask them.

Reply:
It looks like you changed it, but it's definitely possible to receive several messages.